pxr/usd/usd/wrapValidationRegistry.cpp

```diff
@@ -0,0 +1,196 @@
+    {
+        result.append(pointer_wrapper(validator));
+    }
+    return result;
+}
+
+list _GetOrLoadValidatorSuitesByName(UsdValidationRegistry &validationRegistry,
+                                     const TfTokenVector &suiteNames)
+{
+    list result;
+    for (const auto *validatorSuite :
+         validationRegistry.GetOrLoadValidatorSuitesByName(suiteNames))
+    {
+        result.append(pointer_wrapper(validatorSuite));
+    }
+    return result;
+}
+
+object _GetValidatorMetadata(const UsdValidationRegistry &validationRegistry,
+                             const TfToken &name)
+{
+    UsdValidatorMetadata metadata;
+    bool success = validationRegistry.GetValidatorMetadata(name, &metadata);
+    if (success)
+    {
+        return object(metadata);
+    }
+
+    return object();
+}
+
+UsdValidationRegistry &_GetRegistrySingleton(object const & /* classObj */)
+{
+    return UsdValidationRegistry::GetInstance();
+}
+
+object _DummyInit(tuple const & /* args */, dict const & /* kw */)
+{
+    return object();
+}
+
+// Test only for registering validators as no APIs are exposed to register
+// validators from python.
+void _RegisterTestValidators()
+{
+    UsdValidationRegistry &registry = UsdValidationRegistry::GetInstance();
+    {
+        const TfToken validatorName("testValidationPlugin:TestValidator1");
+        const UsdValidateStageTaskFn stageTaskFn =
+            [](const UsdStagePtr &usdStage)
+        {
+            return UsdValidationErrorVector{UsdValidationError(
+                UsdValidationErrorType::Error,
+                {UsdValidationErrorSite(usdStage, SdfPath("/"))},
+                "This is an error on the stage")};
+        };
+
+        // Register the validator
+        TfErrorMark m;
+        registry.RegisterPluginValidator(validatorName, stageTaskFn);
+        TF_AXIOM(m.IsClean());
+    }
+    {
+        const TfToken validatorName("testValidationPlugin:TestValidator2");
+        const UsdValidatePrimTaskFn primTaskFn = [](const UsdPrim & /*prim*/)
+        { return UsdValidationErrorVector{}; };
+
+        // Register the validator
+        TfErrorMark m;
+        registry.RegisterPluginValidator(validatorName, primTaskFn);
+        TF_AXIOM(m.IsClean());
+    }
+    {
+        const TfToken suiteName("testValidationPlugin:TestValidatorSuite");
+        const std::vector<const UsdValidator *> containedValidators =
+            registry.GetOrLoadValidatorsByName(
+                {TfToken("testValidationPlugin:TestValidator1"),
+                 TfToken("testValidationPlugin:TestValidator2")});
+        TfErrorMark m;
+        registry.RegisterPluginValidatorSuite(suiteName, containedValidators);
+        TF_AXIOM(m.IsClean());
+    }
+}
+
+} // anonymous namespace
+
+void wrapUsdValidationRegistry()
+{
+    class_<UsdValidationRegistry, boost::noncopyable>("ValidationRegistry",
+                                                      no_init)
+        .def("__new__", &_GetRegistrySingleton,
```

Comment:
In addition to the TfPySingleton usage comment I have below, the convention for our python singletons is that you just call the class object to get the instance, again referring to UsdSchemaRegistry I have below, notice there is no GetInstance wrapping.